in-market-test/v2/ansible/build_network_file.py

```diff
@@ -0,0 +1,126 @@
+import argparse
+import configparser
+import os
+import re
+import sys
+from pathlib import Path
+
+
+def get_urls_from_config():
+    # Read the Ansible inventory file
+    config = configparser.ConfigParser()
+    config.read(Path("in-market-test/v2/ansible/inventory.ini"))
+
+    hosts = [s.split(" ")[0] for s in list(config["myhosts"].keys())]
+
+    if len(hosts) != 3:
+        print(f"ERROR: Expected 3 hosts in inventory.ini, but found {len(hosts)}")
+        sys.exit(1)
+
+    ssh_config_path = os.path.expanduser("~/.ssh/config")
+    # Read the SSH config file
+    with open(ssh_config_path, "r") as f:
+        ssh_config = f.read()
+    # Parse the SSH config file to get the hostname for each host
+    hostnames = {}
+    hostname_match = re.findall(r"Host (\S+)\n[\s\t]+HostName (\S+)", ssh_config)
+    hostnames = {host: hostname for (host, hostname) in hostname_match if host in hosts}
+
+    missing_hosts = [host for host in hosts if hostnames.get(host) is None]
+    if missing_hosts:
+        print(
+            f"ERROR: ~/.ssh/config is missing hostnames for host: {', '.join(missing_hosts)}"
+        )
+        sys.exit(1)
+
+    return tuple(hostnames[host] for host in hosts)
+
+
+def get_urls():
+    parser = argparse.ArgumentParser(description="Tool to build network.toml from keys")
+    parser.add_argument(
+        "--config",
+        action="store_true",
+        help="Load urls from inventory.ini and ~/.ssh/config",
+    )
+    parser.add_argument(
+        "--helper1-url",
+        required=False,
+        help="URL of helper1",
+    )
+    parser.add_argument(
+        "--helper2-url",
+        required=False,
+        help="URL of helper2",
+    )
+    parser.add_argument(
+        "--helper3-url",
+        required=False,
+        help="URL of helper3",
+    )
+    args = vars(parser.parse_args())
+
+    if args["config"]:
+        return get_urls_from_config()
+    else:
+        missing_args = [arg for (arg, url) in args.items() if url is None]
+        if missing_args:
+            print(
+                "ERROR: If not loading from config, --helper1-url, --helper2-url, "
+                "and, --helper3-url are all required."
+            )
+            sys.exit(1)
+        return (args["helper1_url"], args["helper2_url"], args["helper3_url"])
+
+
+def main():
+
+    urls = get_urls()
+    certs = (
+        Path("in-market-test/v2/deployed_keys/1-cert.pem").read_text(),
+        Path("in-market-test/v2/deployed_keys/2-cert.pem").read_text(),
+        Path("in-market-test/v2/deployed_keys/3-cert.pem").read_text(),
+    )
+    public_keys = (
+        Path("in-market-test/v2/deployed_keys/1-mk.pub").read_text(),
+        Path("in-market-test/v2/deployed_keys/2-mk.pub").read_text(),
+        Path("in-market-test/v2/deployed_keys/3-mk.pub").read_text(),
+    )
+
+    network_template = f"""
```

Comment:
can we put the template on a separate file? like ```network.toml.template```

Reply:
unfortunately not with an f-string

Reply:
Can we use at the very least something like what's described here with strings? https://stackoverflow.com/questions/6385686/is-there-a-native-templating-system-for-plain-text-files-in-python

Reply:
what's the advantage? the f-string makes it very clear what's going where, and also will allow for syntax highlighting if you try to access a variable that doesn't exist.

Reply:
It separates the template from where the values are coming from. The same principle as to why web frameworks separate "views" from controllers/models.

Reply:
plenty of modern web frameworks don't separate those, e.g., React is all about mixing html and JS. 

I might be inclined to take a template route if we had lots of templates, and wanted to be super generic, but here we have a single file to build. it seems like a net-loss to separate the template and the variables, because we no longer get automatic linting that can tell you if you have a typo in one of the variables inserted into the template.